xapian-letor/api/ranker.cc

```diff
@@ -37,11 +37,15 @@
 using namespace std;
 using namespace Xapian;
 
-struct scoreComparer {
-    bool operator()(const pair<Xapian::docid, int>& first_pair, const pair<Xapian::docid, int>& second_pair) const {
-        return first_pair.second > second_pair.second;
-    }
-};
+bool
+Ranker::ScoreComparer::operator()(const FeatureVector & firstfv, const FeatureVector& secondfv) const {
+    return firstfv.get_score() > secondfv.get_score();
+}
```

Comment:
As I understand it, `std::sort` does not allow non-static comparison function. But yes, functors weren't quite needed here.

Pushed them as `static` functions in `ranker.h`. Please review if its done correctly this time or not.